### PR TITLE
Standardize error template paths

### PR DESCRIPTION
**Pull request by agjohnson**
In trying to rework the templates for proxito, I found it quite confusing
keeping the dashboard and proxito templates separate. This separates proxito
specific errors into a separate path just for these templates, `errors/proxito`,
instead of using various paths and naming conventions for these files.

This makes it possible for the new dashboard to isolate proxito error templates
and to extend from a common proxito error base.

- Refs https://github.com/readthedocs/ext-theme/issues/304
- Requires https://github.com/readthedocs/common/pull/220

### CODE REVIEW

**Comment by agjohnson** [comment]
I was expecting this PR to fail miserably, but seems it is either somehow okay or we don't have enough test coverage :shrug:

**Review by stsewd — APPROVED** [review]

File: readthedocs/templates/errors/proxito/404/no_project_page.html [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{% extends "errors/proxito/404/base.html" %}
+{% load i18n %}
+
+{# project exists but page doesn't #}
+
+{% block 404_error_message %}
+  {% include "errors/proxito/404/include_error_message.html" %}
+  {% include "errors/proxito/404/include_search.html" with search_project=project.slug %}
+  {% include "errors/proxito/404/include_tips.html" %}
+{% endblock %}
```
Comment (humitos) [inline readthedocs/templates/errors/proxito/404/no_project_page.html:10]: All these blocks should use `endblock <block name>`.

Comment (agjohnson) [reply readthedocs/templates/errors/proxito/404/no_project_page.html:10]: I didn't alter these templates besides changing paths. If we want to alter these templates, that would be separate work I feel.

And are you thinking of the tests on ext-theme? Are we using djlint on the old templates now? It doesn't seem like a strong priority to be lint fixing our templates we're deprecating either way.

Comment (humitos) [reply readthedocs/templates/errors/proxito/404/no_project_page.html:10]: Oh, no. I don't care about old templates. I thought they were for ext-theme.

**Comment by agjohnson** [comment]
Hrm, seems the last remaining piece is whatever these failures are:

```
------------------------------ Captured log call -------------------------------
WARNING  django.request:log.py:241 I'm a Teapot: /projects/subproject/
____________________ RedirectTests.test_subproject_root_url ____________________

self = <readthedocs.proxito.tests.test_redirects.RedirectTests testMethod=test_subproject_root_url>

    def test_subproject_root_url(self):
        r = self.client.get(
            "/projects/subproject/",
            secure=True,
            headers={"host": "project.dev.readthedocs.io"},
        )
>       self.assertEqual(r.status_code, 302)
E       AssertionError: 418 != 302

readthedocs/proxito/tests/test_redirects.py:89: AssertionError
```

I'm a little confused by them right now. I can see why these tests could fail with the 418 status response, but I can't yet explain _why_.

**Comment by humitos** [comment]
I don't have the answer here, but it seems those tests are hitting the [`dummy_dashboard_urls`](https://github.com/readthedocs/readthedocs.org/blob/7b7e84c348f493c986cf8e343fd856c3de5f2892/readthedocs/proxito/urls.py#L155) instead of the Proxito urls.

**Comment by humitos** [comment]
Maybe the order of the definition changed somehow? `/projects/subproject/` should be caught by https://github.com/readthedocs/readthedocs.org/blob/f941fc73b912ab885075f51d3ea6e6e4da7e03b9/readthedocs/proxito/urls.py#L144

**Comment by humitos** [comment]
Yeap, that's the issue. This diff makes that test to pass again:

```diff
diff --git a/readthedocs/proxito/urls.py b/readthedocs/proxito/urls.py
index ba465b3bc..2f3d33b8c 100644
--- a/readthedocs/proxito/urls.py
+++ b/readthedocs/proxito/urls.py
@@ -212,9 +212,10 @@ debug_urls = [
 groups = [
     health_check_urls,
     proxied_urls,
-    dummy_dashboard_urls,
     core_urls,
     docs_urls,
+    # Dummy dashboard URLs are required to be defined at the end
+    dummy_dashboard_urls,
 ]
 
 if settings.SHOW_DEBUG_TOOLBAR:
```

**Review by humitos — APPROVED** [review]
Looks great! 💯 

I left a comment with the solution for the test that we should include here as well.

**Comment by agjohnson** [comment]
Thanks for that! I figured it had to be something around the URL changes too, I ended my day before getting into debugging at all though.

**Pull request merged**